src/qibotn/backends/quimb.py

```diff
@@ -1,38 +1,77 @@
 from collections import Counter
+from typing import Optional
 
+import quimb as qu
 import quimb.tensor as qtn
-from qibo.backends import NumpyBackend
 from qibo.config import raise_error
-from qibo.result import QuantumState
+from qibo.gates.abstract import ParametrizedGate
+from qibo.models import Circuit
 
 from qibotn.backends.abstract import QibotnBackend
 from qibotn.result import TensorNetworkResult
 
+GATE_MAP = {
+    "h": "H",
+    "x": "X",
+    "y": "Y",
+    "z": "Z",
+    "s": "S",
+    "t": "T",
+    "rx": "RX",
+    "ry": "RY",
+    "rz": "RZ",
+    "u3": "U3",  # TODO: check
+    "cx": "CX",
+    "cnot": "CNOT",
+    "cy": "CY",
+    "cz": "CZ",
+    "iswap": "ISWAP",
+    "swap": "SWAP",
+    "ccx": "CCX",
+    "ccy": "CCY",
+    "ccz": "CCZ",
+    "toffoli": "TOFFOLI",
+    "cswap": "CSWAP",
+    "fredkin": "FREDKIN",
+    "fsim": "fsim",
+    "measure": "measure",
+}
 
-class QuimbBackend(QibotnBackend, NumpyBackend):
 
-    def __init__(self):
-        super().__init__()
+if not __name__ == "__main__":
+
+    def __init__(self, quimb_backend="numpy", contraction_optimizer="auto-hq"):
```

Comment:
Why are you adding an `__init__` method to something which is not a class any longer?

At this point, I guess the `class` definition is actually just missing.